### DISCUSSION TITLE
Olimex kein Ethernet

### DISCUSSION THREAD

**Discussion by Troubadix** [tbnobody/OpenDTU#2279] [General]
Hallo, 

habe einen Olimex PoE samt Display bisher mit NRF Modul und jetzt umgebaut auf CMT. 
Hat früher super funktioniert, jetzt geht Ethernet nicht mehr. Nur noch WLAN. Die Kommunikation zum Wechselrichter geht aber. 

Pin Config für ETH ist identisch zu vorher. 
Wo liegt mein Fehler? Danke. 

![Screenshot_20240914-120745](https://github.com/user-attachments/assets/d07842c5-d486-4e75-bee1-23ebfe319441)
![Screenshot_20240914-120738](https://github.com/user-attachments/assets/0553294d-7f82-4a2a-9ecf-c80088877974)

**Comment by ghost** [comment]
Hi,
keine Ahnung.
Was auffällig ist, du benutzt teilweise andere Pins für das CMT, als die, die durch den Wegfall des NRF24 frei geworden sind.
Vielleicht biegst du den SPI Port ungünstig oder zu sehr um.
Ich würde eine Pin-Belegung am CMT versuchen, die der des NRF nahe kommt:
clk: 18
sc: 5
fcs: 4
sdio: 19 oder 23
gpio2: -1
gpio3: -1
gpio2 und gpio3 werden wohl nicht benötigt.

Möglicherweise helfen dir [dieser Beitrag](https://github.com/tbnobody/OpenDTU/discussions/873) oder [dieser](https://github.com/tbnobody/OpenDTU/discussions/1041).

(Was mich wundert ist, das deine Konstellation mit NRF funktioniert hat, da Clock am NRF Pin 18 war und  der Daten Pin für Ethernet aber ebenfalls Pin 18?)

**Comment by Troubadix** [comment]
Hi, 

die Config in der json fürs NRF ist ja weggefallen - das was da steht ist das, was wohl einkompiliert(?) ist?

Das hier war die Config fürs NRF:
![grafik](https://github.com/user-attachments/assets/5bbc31a7-9f9c-42e3-bfaa-24c53b5bf9b0)

**Reply by ghost** [reply]
> das was da steht ist das, was wohl einkompiliert(?) ist?

Rechts steht, was benutzt wird.
In der Mitte, das was in der momentan geladenen pinnmapping.json. steht; falls du eine neue Konfiguration ausgewählt und noch nicht übernommen hast, weicht das ab, sonst ist das mit den rechten Werten identisch.

**Comment by Troubadix** [comment]
Hab jetzt mal 
        "cmt": {
            "clk": 14,
            "cs": 5,
            "fcs": 13,
            "sdio": 16,
            "gpio2": -1,
            "gpio3": -1
},
probiert, gleiches Ergebnis. Kommunikation zum WR geht, LAN geht nicht.

**Comment by ghost** [comment]
Geht Ethernet allgemein überhaupt noch, ohne das CMT?
Nicht das sich da etwas am Olimex verabschiedet hat.
Knipse das CMT mal testweise aus, sprich eine Config ohne CMT auswählen.
Verwendest du vielleicht die selbe IP für LAN und WLAN? Könnte auch ein Netzwerkproblem sein.

In den von mir verlinkten Beiträgen, mit ebenfalls Ethernet Problemen scheint das Nachfolgende zu funktionieren, sogar mit CMT+NRF zusammen.
Den NRF Teil kannst du auch entfernen. Wenn das nicht hilft, weiß ich auch nicht weiter.
Dieser Satz wird nicht nötig sein, aber sicherheitshalber: die Pins sind dann auch so wie angegeben am ESP zu verdrahten.

 "name": "Olimex ESP32-POE CMT2003a + NRF",
    "cmt": {
        "clk": 15,
        "cs": 16,
        "fcs": 13,
        "sdio": 2
    },
    "nrf24": {
        "miso": 36,
        "mosi": 32,
        "clk": 4,
        "irq": 35,
        "en": 14,
        "cs": 5
    },
    "eth": {
        "enabled": true,
        "phy_addr": 0,
        "power": 12,
        "mdc": 23,
        "mdio": 18,
        "type": 0,
        "clk_mode": 3
    },
    "led": {
        "led0": -1,
        "led1": -1
    },
    "display": {
        "type": 3,
        "data": 33,
        "clk": 0
    }

**Comment by Troubadix** [comment]
Jetzt gehts. Das NRF24 muss komplett auf -1 gesetzt werden.

Dein letztes Beispiel hat mich auf den richtigen Weg gebracht - das ging bis aufs Display, welches bei mir schon fest verlötet ist mit clk=32. Da fiel mir dann die Doppelung zum nrf auf, also da alles -1 und schon passt es.

Danke!!! :)
